packages/cli/src/Server.ts

```diff
@@ -2898,6 +2898,16 @@ export async function start(): Promise<void> {
 
 		void InternalHooksManager.getInstance().onServerStarted(diagnosticInfo);
 	});
+
+	// @ts-ignore
+	server.on('error', (error) => {
+		if (error.code === 'EADDRINUSE') {
+			console.log(
+				`n8n's port ${PORT} is already in use. Do you have another instance of n8n running already?`,
+			);
+			process.exit(0);
```

Comment:
```suggestion
			process.exit(1);
```

Reply:
Other cli requirements also exit in code `0` ( https://github.com/n8n-io/n8n/blob/6453996de30e9635bda373f9f120a54621a8a69f/packages/cli/bin/n8n#L29)
Should we update them as well?

Reply:
Afaik non-zero exit signifies failure. Cannot say why the other checks don't use it.

Reply:
as discussed with @krynble , updating other's as well